src/liballoc/raw_vec.rs

```diff
@@ -194,7 +195,9 @@ impl<T, A: Alloc> RawVec<T, A> {
     /// Unique::empty() if `cap = 0` or T is zero-sized. In the former case, you must
     /// be careful.
     pub fn ptr(&self) -> *mut T {
```

Comment:
Should this just return a `NonNull<T>`?

Reply:
Maybe? There might be a reason for why this looks the way it does. Possibly because `NonNull` is not very ergonomic.

Reply:
yeah, NonNull is only useful as a storage type, it's useless for doing actual work.